lib/dialogs/settings.jsx

```diff
@@ -37,6 +38,74 @@ export const SettingsDialog = React.createClass({
     });
   },
 
+  onSignOutRequested() {
+    // Safety first! Check for any unsynced notes before signing out.
+    const { onSignOut, noteBucket } = this.props;
+    const { notes } = this.props.appState;
+    const { getVersion } = noteBucket;
+
+    if (
+      noteBucket.hasLocalChanges((error, hasChanges) => {
+        if (hasChanges) {
+          this.showUnsyncedWarning();
+          return;
+        }
+
+        // Also check persisted store for any notes with version 0
+        const noteHasSynced = note =>
+          new Promise((resolve, reject) =>
+            getVersion(note.id, (e, v) => (e || v === 0 ? reject() : resolve()))
+          );
+
+        Promise.race(notes.map(noteHasSynced)).then(
```

Comment:
Should this be `Promise.all` not `Promise.race`? You want to make sure that none of  the notes are unsynced right?